File: LeetCode/11.盛最多水的容器.cs
public class Solution
{
    public int MaxArea(int[] height)
    {
        if (height.Length < 2) return -1;
        int i = 0, j = height.Length - 1, res = 0;
        while (i < j)
        {
            int h =  height[i] <= height[j] ? height[i] : height[j];
            //if((j - i) * h > res)res = (j - i) * h;
            res = (j - i) * h > res ? (j - i) * h : res;
            if (height[i] < height[j]) i++;
            else j--;
        }
        return res;
    }
}

File: LeetCode/11.盛最多水的容器.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i = 0
        j = len(height)-1
        res = 0;
        if j < 1: return -1;
        while(i < j):
            h = min(height[i],height[j])
            res = h*(j-i) if h*(j-i) > res else res
            if height[i] < height[j]:i+=1
            else: j-=1
        return res